### ISSUE TITLE
Add optimizer wrapper solution

### ISSUE THREAD

**Issue by andreped** [andreped/GradientAccumulator#27]
It was observed that model wrapping, where we overload the `train_step`, is not compatible with distribute training strategies (multi-GPU training), as discussed in [this thread](https://github.com/keras-team/keras/issues/17429#issuecomment-1405612981).

What should work better is to wrap the optimizer instead, as we should have better control of the gradients across replicates, and can define the correct behaviour.

**Comment by andreped** [comment]
Mostly fixed in df2ad165e29da1974a0909a5f9dd55f0d65753cc.

However, there are some issues with using other optimizers. Nonetheless, we reach expected results (compared to regular batch training), using the SGD optimizer.

We should make a separate issue regarding that.
